src/mongocxx/write_concern.cpp

```diff
@@ -65,9 +65,27 @@ void write_concern::nodes(std::int32_t confirm_from) {
 }
 
 void write_concern::acknowledge_level(write_concern::level confirm_level) {
-    libmongoc::write_concern_set_w(
-        _impl->write_concern_t,
-        static_cast<std::underlying_type<write_concern::level>::type>(confirm_level));
+    std::int32_t w;
+    switch (confirm_level) {
+    case write_concern::level::k_default:
+        w = MONGOC_WRITE_CONCERN_W_DEFAULT;
+        break;
+    case write_concern::level::k_majority:
+        w = MONGOC_WRITE_CONCERN_W_MAJORITY;
+        break;
+    case write_concern::level::k_unacknowledged:
+        w = MONGOC_WRITE_CONCERN_W_UNACKNOWLEDGED;
+        break;
+    case write_concern::level::k_tag:
+        // no exception for setting tag if it's set
+        if (libmongoc::write_concern_get_w(_impl->write_concern_t) == MONGOC_WRITE_CONCERN_W_TAG)
+            return;
+    default:
+        throw std::invalid_argument(
```

Comment:
Not sure if this is your question... but this is reached when calling `wc.acknowledge_level(k_unknown)` or `wc.acknowledge_level(k_tag)` (but I allow setting it to tag if a tag is already specified, I thought it would be nice to have the option of being explicit).